akka-actor/src/main/scala/akka/serialization/Serialization.scala

```diff
@@ -105,8 +106,34 @@ class Serialization(val system: ExtendedActorSystem) extends Extension {
     }
 
   /**
+   * Deserializes the given array of bytes using the specified serializer id,
+   * using the optional type hint to the Serializer and the optional ClassLoader ot load it into.
+   * Returns either the resulting object or an Exception if one was thrown.
+   */
+  def deserialize[T](bytes: Array[Byte], serializerId: Int, manifest: String): Try[T] =
+    Try {
+      val serializer = try serializerByIdentity(serializerId) catch {
+        case _: NoSuchElementException ⇒ throw new NotSerializableException(
+          s"Cannot find serializer with id [$serializerId]. The most probable reason is that the configuration entry " +
+            "akka.actor.serializers is not in synch between the two systems.")
+      }
+      serializer match {
+        case s2: SerializerWithStringManifest ⇒ s2.fromBinary(bytes, manifest).asInstanceOf[T]
+        case s1 if manifest == "" || (manifest eq null) ⇒
+          s1.fromBinary(bytes, None).asInstanceOf[T]
```

Comment:
AFAICS the `manifest eq null` shouldn't really happen in normal operation, as we `if !hasManifest ""` in other places. It's OK to keep the null case here, just making sure that it's not expected due something I missed